src/solver/optimisation/adequacy_patch_csr/solve_problem.cpp

```diff
@@ -55,9 +55,10 @@ extern "C"
 
 using namespace Antares;
 
-void setInteriorPointProblem(PROBLEME_ANTARES_A_RESOUDRE* ProblemeAResoudre,
-                             PROBLEME_POINT_INTERIEUR& Probleme)
+std::unique_ptr<PROBLEME_POINT_INTERIEUR> buildInteriorPointProblem(
+  PROBLEME_ANTARES_A_RESOUDRE* ProblemeAResoudre)
 {
+    std::unique_ptr<PROBLEME_POINT_INTERIEUR> Probleme(new PROBLEME_POINT_INTERIEUR());
```

Comment:
```suggestion
    auto Probleme = std::make_unique<PROBLEME_POINT_INTERIEUR>();
```

You need to merge branch `develop` for that, since it contains #779. This feature is only available from C++17.